tests/disjoint2.mm

```diff
@@ -0,0 +1,20 @@
+$c var formula |- ( ) $.
+
+$v x y z w $.
+xf $f formula x $.
+yf $f formula y $.
+zf $f formula z $.
+wf $f formula w $.
+combo $a formula ( x y ) $.
+${
+  $d x y $.
+  ax-1 $a |- ( x y ) $.
+$}
+
+$( Uncomment either good or bad to trigger an infinite loop
+   in the distinct variable checker. $)
+$( good $p |- ( ( x y ) ( z w ) ) $=
+  xf yf combo zf wf combo ax-1 $. $)
```

Comment:
Is the name `good` indicative? Because I would have assumed that this is a DV failure - none of `x,y,z,w` are disjoint.

Reply:
I realized that after I wrote it. You are correct that to be truly good this would need some $d added.

Reply:
OK, I apparently couldn't resist doubling down on this naming convention so the test now has `verybad` , `stillbad` , `semigood` , `almostgood` , and `good` .